### ISSUE TITLE
loss improves consistently but turns to NaN at some point

### ISSUE THREAD

**Issue by Avi-avidan** [matterport/Mask_RCNN#458]
Could you advise on this issue (see image attached)?
![image](https://user-images.githubusercontent.com/25052915/38916775-503662da-42f1-11e8-8815-b73e6c50af4e.png)

**Comment by waleedka** [comment]
The `loss` is the sum of all the other losses. My guess is that one of the other losses went to NaN and caused the total loss to be NaN as well. Can you show all the other losses?

**Comment by liangbo-1** [comment]
Have you solved your problem？

**Comment by Avi-avidan** [comment]
Unfortunately, I didn't get to it.
Mostly because I needed to segment high res images (1280x960) and couldn't train this high res mcrnn on my GTX1080 (11GB).

Since loss was improving consistently, my guess is one of the losses was turning zero (or very very close to it) and that way there was no gradient at that point.

A simple quick solution would be to add a constant epsilon. I did implement this in a different project. keras example:

import keras.backend as K

def euclidean_distance_loss(y_true, y_pred, epsilon=1e-9):
    return K.sum(K.sqrt(K.sum(K.square(y_pred - y_true), axis=-1) + epsilon), axis=-1)


let me know if this helped you.

Best,
Avi
